backend/src/api.py

```diff
@@ -119,6 +121,81 @@ def inner_wrapper(wrapped_func: T) -> T:
+            # We don't want to run this check on iterator helpers as they can have different input/output metadata than what they actually run
+            if ann is not None and node_type != "iteratorHelper":
+                # Pop the return type key off the dict
+                return_type = ann.get("return", None)
+                if return_type is not None:
+                    return_type = str(return_type)
+                    if return_type.startswith("Tuple"):
+                        # Evaluate the return type
+                        # pylint: disable=eval-used
+                        evaluated_return_type = eval(
+                            return_type, wrapped_func.__globals__
+                        )
+                        output_len = len(get_args(evaluated_return_type))
+                    elif return_type == "None":
+                        output_len = 0
+                    else:
+                        output_len = 1
+                    if output_len != len(p_output):
+                        logger.warning(
+                            f"Number of outputs and return types don't match for {schema_id}"
+                        )
+                # Remove the return type from the annotations
+                ann.pop("return", None)
+                input_len = len(ann.keys())
+                # Iterators pass in their context, so we need to account for that
+                if node_type == "iterator":
+                    input_len -= 1
+                # Variable args don't have good typing, so right now we just hardcode what to ignore.
+                if list(ann.keys())[-1] not in ["args", "sources"] and input_len != len(
+                    p_inputs
+                ):
+                    logger.warning(
+                        f"Number of inputs and annotations don't match for {schema_id}: {input_len=} != {len(p_inputs)=}"
+                    )
+                for (py_var, py_type), input_class in zip(ann.items(), p_inputs):
+                    associated_type = input_class.associated_type
+                    if associated_type is not None:
+                        # logger.info(f"{py_type=} {associated_type=}")
+                        evaluated_py_type = py_type
+                        if isinstance(py_type, str):
+                            try:
+                                # Allows us to use pipe unions still
+                                if "|" in py_type:
+                                    py_type = f'Union[{py_type.replace(" |", ",")}]'
+                                global_vals = wrapped_func.__globals__
+                                # Gotta add these to the scope
+                                global_vals.update({"Union": Union})
+                                global_vals.update({"np": np})
+                                # pylint: disable=eval-used
+                                evaluated_py_type = eval(py_type, global_vals)
```

Comment:
Could we use locals instead of changing global scope? https://docs.python.org/3/library/functions.html#eval